### PR TITLE
Use mongomock for testing

### PR DESCRIPTION
**Pull request by abulte**
Ref http://docs.mongoengine.org/guide/mongomock.html

This should speed up testing (twice as fast on my machine). 

:warning: remove `MONGODB_HOST_TEST` from your local `udata.cfg`.

Fix https://github.com/etalab/data.gouv.fr/issues/467

### CODE REVIEW

**Comment by abulte** [comment]
Unfortunately we still have two tests failing although I rewrote them and got rid of technical debt in the mean time :-/ Might be worth opening an issue on `mongomock`.

**Comment by abulte** [comment]
Other refactoring idea: https://github.com/opendatateam/udata/blob/0758d23d3fff92d6f61dbb4f7666142a7e2bef0d/udata/core/reuse/api.py#L171 — followers also uses `GenericReferenceField` but thanks to the `model` abstraction, it always uses a full-fledged object to make the query, instead of an id (which is the problem with our two failing tests).

**Comment by maudetes** [comment]
I think it is worth pushing this refactoring through to save time for testing, and additionally removing some technical debt.

File: udata/settings.py [hunk 1]
```diff
@@ -494,6 +494,7 @@ class Testing(object):
         'title^2',
         'description',
     )
+    MONGODB_HOST_TEST = 'mongomock://localhost:27017/udata'
```
Comment (maudetes) [inline udata/settings.py:497]: Maybe update docs (`adapting-settings.md` and `testing-code.md`) with mongomock changes.

**Comment by maudetes** [comment]
Issue created by @quaxsze with reproductible bug: https://github.com/mongomock/mongomock/issues/719

**Comment by maudetes** [comment]
Closing for now, let's reopen it if the mongomock ticket gets resolved.

**Pull request closed without merging**